src/components/PrimaryNav/PrimaryNavItem/PrimaryNavItem.jsx

```diff
@@ -62,23 +73,35 @@ ItemTypeResolver.propTypes = {
     button: PropTypes.node
 };
 
-export const PrimaryNavItem = ({label, icon, className, subtitle, url, button, isSelected, onClick, badge, ...props}) => (
-    <li
-        className={classnames(
-            styles.primaryNavItem,
-            {[styles.selected]: isSelected},
-            className
-        )}
-        title={label}
-        onClick={onClick}
-        {...props}
-    >
-        <ItemTypeResolver icon={icon} label={label} subtitle={subtitle} url={url} button={button}/>
-
-        {badge &&
-            <badge.type className={classnames(styles.primaryNavItem_badge)} color="danger" type="round" label={badge.props.label}/>}
-    </li>
-);
+export const PrimaryNavItem = ({label, icon, className, subtitle, url, button, isSelected, onClick, badge, ...props}) => {
+    const primaryNavContext = useContext(PrimaryNavContext);
+
+    return (
+        <li
+            className={classnames(
+                styles.primaryNavItem,
+                {[styles.selected]: isSelected},
+                className
+            )}
+            title={label}
+            onClick={e => {
+                primaryNavContext.collapse();
+                if (onClick) {
```

Comment:
You can add this, to manage empty onClick props
```
PrimaryNavItem.defaultProps = {
    onClick: () => {}
}
```

Reply:
good point